int32 cutoff_freq